CMakeLists.txt

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.17...3.18 FATAL_ERROR)
```

Comment:
There hasn't been a need to pin the upper version in the past here, is there a reason we should pin to `>=3.17,<=3.18`?

Reply:
Despite the way it looks (it confused me, too, when I first saw this), it doesn't actually set a maximum version. What it does, basically, do this:

When cmake changes behavior, it does so via policies, which are set to "old" by default based on the cmake version expected. What this does is to set all policies that were introduced up to 3.18 are set to "new", so this basically expresses that we require at least 3.17, but it keeps working with cmake new policies in 3.18, too.

Specifically, there's https://cmake.org/cmake/help/latest/policy/CMP0104.html in 3.18, which otherwise leads to a warning about `CUDA_ARCHITECTURES` being empty, which used to be allowed. The new behavior sets `CUDA_ARCHITECTURES` to `CMAKE_CUDA_ARCHITECTURES`, which is initialized to the lowest arch supported by nvcc by default, but can be of course be overridden as needed.